cdap-security/src/main/java/io/cdap/cdap/security/tools/AccessTokenGeneratorService.java

```diff
@@ -0,0 +1,216 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.tools;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import io.cdap.cdap.api.metrics.MetricsCollectionService;
+import io.cdap.cdap.common.BadRequestException;
+import io.cdap.cdap.common.HttpExceptionHandler;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.conf.SConfiguration;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.common.guice.IOModule;
+import io.cdap.cdap.common.http.CommonNettyHttpServiceFactory;
+import io.cdap.cdap.common.io.Codec;
+import io.cdap.cdap.common.metrics.NoOpMetricsCollectionService;
+import io.cdap.cdap.common.security.HttpsEnabler;
+import io.cdap.cdap.security.auth.AccessToken;
+import io.cdap.cdap.security.auth.TokenManager;
+import io.cdap.cdap.security.auth.UserIdentity;
+import io.cdap.cdap.security.guice.CoreSecurityRuntimeModule;
+import io.cdap.http.AbstractHttpHandler;
+import io.cdap.http.HandlerContext;
+import io.cdap.http.HttpResponder;
+import io.cdap.http.NettyHttpService;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Base64;
+import java.util.Collections;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+/**
+ * A service for managing the lifecycle of an http service that serves access tokens.
+ */
+public class AccessTokenGeneratorService extends AbstractIdleService {
+  static final String ADDRESS_DEFAULT = "127.0.0.1";
+  static final int PORT_DEFAULT = 11030;
+  // HttpHandler has only one caller, and is accessed infrequently so one worker thread is sufficient
+  static final int WORKER_POOL_SIZE_DEFAULT = 1;
+  static final int EXEC_POOL_SIZE_DEFAULT = 0;
+  private final NettyHttpService httpService;
+  private final TokenGeneratorHttpHandler handler;
+
+  AccessTokenGeneratorService(String address, int port, int workerThreadPoolSize, int execThreadPoolSize) {
+    CConfiguration cConf = CConfiguration.create();
+    Injector injector = Guice.createInjector(
+      new ConfigModule(),
```

Comment:
pass the cConf created above to create the ConfigModule

Reply:
done